src/containers/app/CreateApp.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FormControl, FormGroup, ControlLabel, Button, Alert } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import AsyncContent, { AsyncStatePropType } from '../../components/asynccontent/AsyncContent';
+import { createAppRequest } from './AppActionFactory';
+
+
+class CreateApp extends React.Component {
+
+  static propTypes = {
+    actions: PropTypes.shape({
+      onCreateApp: PropTypes.func.isRequired
+    }).isRequired,
+    createAppAsyncState: AsyncStatePropType.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      name: '',
+      appTypeIds: [],
+      url: ''
+    };
+  }
+
+  componentDidMount() {
+  }
+
+  onTitleChange = (event) => {
+    this.setState({
+      title: event.target.value
+    });
+  };
+
+  onNameChange = (event) => {
+    this.setState({
+      name: event.target.value
+    });
+  };
+
+  onDescriptionChange = (event) => {
+    this.setState({
+      description: event.target.value
+    });
+  };
+
+  onUrlChange = (event) => {
+    this.setState({
+      url: event.target.value
+    });
+  };
+
+  onAppTypeIdsChange = (event) => {
+    this.setState({
+      appTypeIds: event.target.value
+    });
+  };
+
+  onSubmit = () => {
+    const { title, description, name, appTypeIds, url } = this.state;
+    // need to separate the appTypeIds string by comma, put in array
+    const splitAppTypeIds = appTypeIds.split(', ');
+
+    const App = {
+      name,
+      title,
+      description,
+      appTypeIds: splitAppTypeIds,
+      url
+    };
+
+    this.props.actions.onCreateApp(App);
+  }
+
+  renderPending = () => {
+    return (
+      <form onSubmit={this.onSubmit}>
+        <FormGroup>
+          <ControlLabel>Name</ControlLabel>
+          <FormControl type="text" onChange={this.onNameChange} />
+        </FormGroup>
+
+        <FormGroup>
+          <ControlLabel>Title</ControlLabel>
+          <FormControl type="text" onChange={this.onTitleChange} />
+        </FormGroup>
+
+        <FormGroup>
+          <ControlLabel>Description</ControlLabel>
+          <FormControl componentClass="textarea" onChange={this.onDescriptionChange} />
+        </FormGroup>
+
+        <FormGroup>
+          <ControlLabel>App Type Ids</ControlLabel>
+          <FormControl componentClass="textarea" onChange={this.onAppTypeIdsChange} placeholder="idNumberOne, idNumberTwo, idNumberThree" />
```

Comment:
I don't recommend using a textarea to store a single string representing all ids.

Reply:
Plan to use a Multiple Select when I get the API endpoint for getAllAppTypes. Leaving as is for now.